scylla/src/policies/load_balancing/single_target.rs

```diff
@@ -0,0 +1,90 @@
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+use uuid::Uuid;
+
+use crate::cluster::{ClusterState, Node, NodeRef};
+use crate::routing::Shard;
+
+use super::{LoadBalancingPolicy, RoutingInfo};
+
+/// Node identifier used by [`SingleTargetLoadBalancingPolicy`].
+///
+/// If `NodeIdentifier::Node` is used, it will be returned immediately without
+/// the lookup in the cluster metadata.
+///
+/// If `NodeIdentifier::HostId` or `NodeIdentifier::NodeAddress` is used,
+/// the node will be looked up in the cluster metadata based on its host_id/address.
+///
+/// Note: `NodeIdentifier::NodeAddress` assumes that provided address is **untranslated**.
+/// In other words, it should be the address of form `<rpc_address>:<cql_port>`, where
+/// `<rpc_address>` is the address of the node stored in the system tables, as well as
+/// in the [`Node`] struct.
+#[derive(Debug, Clone)] // <- Cheaply clonable
+#[non_exhaustive]
+pub enum NodeIdentifier {
+    Node(Arc<Node>),
+    HostId(Uuid),
+    NodeAddress(SocketAddr),
+}
+
+/// Load balancing policy that enforces a single target.
+///
+/// It may be useful for queries to node-local system tables.
+#[derive(Debug)]
+pub struct SingleTargetLoadBalancingPolicy {
+    node_identifier: NodeIdentifier,
+    shard: Option<Shard>,
+}
+
+impl SingleTargetLoadBalancingPolicy {
+    pub fn new(node_identifier: NodeIdentifier, shard: Option<Shard>) -> Self {
+        Self {
+            node_identifier,
+            shard,
+        }
+    }
+}
```

Comment:
Wouldn't it be easier for the user if `new` returned `Arc`? All the use cases of this struct that can exist require wrapping it in Arc anyway, right?

If there ever emerges use case for creating this struct without Arc then we can add `new_raw` method.

`DefaultPolicyBuilder::build` returns `Arc<dyn LoadBalancingPolicy>`, probably for the same reason.